### PR TITLE
Dev databoxedge microsoft.data box edge 2020 07 01 preview

### PR DESCRIPTION
**Pull request by pvsraviteja**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
API change:
https://github.com/Azure/azure-rest-api-specs/pull/10044/files/8de2e9debff7b404dd711b70b1b8159f98e0c2e2..b73c6ee8b1b65590bb331cedba4286411584b6d8

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json [hunk 1]
```diff
@@ -136,6 +136,52 @@
...
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "description": "Specify $filter='location eq <location>' to filter on location resource or Specify $skipToken=<token> to populate the next page in the list.",
```
Comment (weidongxu-microsoft) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:156]: It is `$expand`, why description is about filter and skipToken?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json [hunk 2]
```diff
@@ -136,6 +136,52 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```
Comment (weidongxu-microsoft) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:176]: `DataBoxEdgeSkuList` has `nextLink` property. Is it correct here as `null`?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json [hunk 3]
```diff
@@ -3920,6 +3979,31 @@
           "description": "The number of nodes in the cluster.",
           "type": "integer",
           "readOnly": true
+        },
+        "resourceMoveDetails": {
```
Comment (weidongxu-microsoft) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:3983]: Is it `readonly`?

Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:3983]: Yes changed it.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json (hunk 2 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:176]: This API is not used. But changed it.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json (hunk 1 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json]: Changed it.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by majastrz** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/getDCAccessCode": {

if you call it listDCAccessCode, it will integrate nicely with ARM templates.

---
Refers to: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:951 in 7546fc3. [](commit_id = 7546fc3a4534b0679eaa308bef1d29faf380d78b, deletion_comment = False)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side with a comment. If getting the access code doesn't make sense in a template, it's fine as is.

**Comment by weidongxu-microsoft** [comment]
@pvsraviteja

Please check ARM's feedback and see if you need to act on it.

When you are all done, let me know when you want the PR merged.

**Comment by pvsraviteja** [comment]
> Signed off from ARM side with a comment. If getting the access code doesn't make sense in a template, it's fine as is.

Getting accesscode is an action and it doesn't fit with ARM template I think

**Comment by pvsraviteja** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/getDCAccessCode": {
> > ```
> 
> if you call it listDCAccessCode, it will integrate nicely with ARM templates.
> 
> Refers to: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-07-01-preview/databoxedge.json:951 in [7546fc3](https://github.com/Azure/azure-rest-api-specs/commit/7546fc3a4534b0679eaa308bef1d29faf380d78b). [](commit_id = [7546fc3](https://github.com/Azure/azure-rest-api-specs/commit/7546fc3a4534b0679eaa308bef1d29faf380d78b), deletion_comment = False)

I think GetDCAccessCode doesn't fit with ARM Template.

**Comment by majastrz** [comment]
> I think GetDCAccessCode doesn't fit with ARM Template.

ARM templates allow using the `listKeys()` function to access the keys of a storage accounts - it is a very common scenario in ARM templates. (Behind the scenes, it translates to a POST request to the RP.) This is somewhat documented at https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list. 

Renaming an action in the future will be a breaking change which requires separate approvals these days. As such, it's better to get those changes in as early as possible to avoid it. Leaving it up to you, though.

**Comment by weidongxu-microsoft** [comment]
@pvsraviteja
Let me know when you plan to merge the PR.

**Comment by weidongxu-microsoft** [comment]
@pvsraviteja
Let me know when you plan to merge the PR.

Also there is now conflict in readme.md

**Comment by pvsraviteja** [comment]
Closing this as it's not relevant.

**Pull request closed without merging**